document/CoreFormat.mdk

```diff
@@ -1095,99 +1123,78 @@ fg
-
-When a region object represents a region in a text file, and the first and last characters in the region occur on the same line,
-the `endLine` property shall not be present.
+When a `region` object represents a region in a text file,
+it may contain a property named `endLine` which shall have an integer value equal to
+the line number of the line containing the last character in the region.
 
-When a `region` object represents a region in a binary file, the `endLine` property shall not be present.
+If `endLine` is absent, it shall be inferred as specified in &sect;[#region-textFiles].
 
 ### `endColumn` property { #region-endColumn }
 
-When a `region` object represents a region in a text file, it may contain a property named `endColumn`.
-If the region has a length of 0, then `endColumn` shall not be present.
-If the region has a length greater than 0, then `endColumn` shall be present
-(even if its value is the same as the value of `startColumn`,
-which could happen if the first and last characters in the range are on different lines),
-and its value shall be the column number of the last character of the region.
-
-When a `region` object represents a region in a binary file, the `endColumn` property shall not be present.
+When a `region` object represents a region in a text file,
+it may contain a property named `endColumn` which shall have an integer value equal to
+the column number of the last character in the region.
 
-### `charOffset` property { #region-charOffset }
+If `endColumn` is absent, it shall be inferred as specified in  &sect;[#region-textFiles].
 
-When a `region` object represents a region in a text file, it may contain a property named `charOffset`,
-which shall have a non-negative integer value equal to the offset (measured in characters) of the first character in the region
-from the beginning of the file.
+### `offset` property { #region-offset}
 
-The offset of the first character in a file shall have the value 0.
+When a `region` object represents a region in a binary file,
+it shall contain a property named `offset` which shall have an integer value equal to
+the byte offset from the beginning of the file of the first byte in the region.
 
-When a `region` object represents a region in a binary file, the `charOffset` property shall not be present.
-
-### `byteOffset` property { #region-byteOffset}
-
-When a `region` object represents a region in a binary file, it shall contain a property named `byteOffset`,
-which shall have an integer value equal to the byte offset of the first character in the region
-from the beginning of the file.
-
-The offset of the first byte in a file shall have the value 0.
-
-When a `region` object represents a region in a text file, the `byteOffset` property shall not be present.
+When a `region` object represents a region in a text file, the `offset` property shall not be present.
```

Comment:
Again, not sure this restriction helps us. We have clearly specified what a column consists of in the spec now (including details around surrogate pairs). For this case, couldn't a SARIF producer emit an offset that conforms to the conditions driving column #'s? So, the spec would say, you must compute an explicit startLine in order to indicate this is a text file (and you must provide at least one of length or endColumn in order to indicate a range). Otherwise, the producer may or may not have populated other orienting values.